src/LayerPlan.cpp

```diff
@@ -849,8 +849,13 @@ void LayerPlan::writeGCode(GCodeExport& gcode)
             if (path.config->isTravelPath())
                 speed *= extruder_plan.getTravelSpeedFactor();
             else
+            {
                 speed *= extruder_plan.getExtrudeSpeedFactor();
 
+                // for some movements such as prime tower purge, the speed may get changed by this factor
+                speed *= path.config->getExtrusionSpeedFactor();
```

Comment:
In the scope of this feature, this speed factor should only affect extrusion travel speeds, not pure travel moves. It is possible to extend this factor to be used for both cases, but it is not intended at the moment. I think changing the name to speed factor to `speed_factor` makes it too general and implies that it does something to the travel move too, which it doesn't. I think we should leave this to when we want to extend this feature.